src/main/java/com/blackducksoftware/integration/hub/notification/content/detail/NotificationContentDetailFactory.java

```diff
@@ -0,0 +1,123 @@
+/**
+ * hub-common
+ *
+ * Copyright (C) 2018 Black Duck Software, Inc.
+ * http://www.blackducksoftware.com/
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.blackducksoftware.integration.hub.notification.content.detail;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import com.blackducksoftware.integration.hub.api.generated.enumeration.NotificationType;
+import com.blackducksoftware.integration.hub.api.view.CommonNotificationState;
+import com.blackducksoftware.integration.hub.notification.content.NotificationContent;
+import com.blackducksoftware.integration.hub.notification.content.PolicyOverrideNotificationContent;
+import com.blackducksoftware.integration.hub.notification.content.RuleViolationClearedNotificationContent;
+import com.blackducksoftware.integration.hub.notification.content.RuleViolationNotificationContent;
+import com.blackducksoftware.integration.hub.notification.content.VulnerabilityNotificationContent;
+
+public class NotificationContentDetailFactory {
+    public List<NotificationContentDetail> generateNotificationContentDetails(final CommonNotificationState commonNotificationState) {
```

Comment:
This should take a view and create NotificationContent before generating the details. @psantos1113